pkg/volume/csi/csi_util.go

```diff
@@ -140,3 +148,66 @@ func hasReadWriteOnce(modes []api.PersistentVolumeAccessMode) bool {
 	}
 	return false
 }
+
+// SanitizeMsg scans proto message for map[string]string marked with csi_secret
+// amd replaces key's value with "* * * Sanitized * * *"
+func SanitizeMsg(pb interface{}) string {
+	if _, ok := pb.(descriptor.Message); !ok {
+		return ""
+	}
+	_, md := descriptor.ForMessage(pb.(descriptor.Message))
+	fields := md.GetField()
+	if fields == nil {
+		return ""
+	}
+	sanitizeFields := []descr.FieldDescriptorProto{}
+	for _, field := range fields {
+		opt, err := proto.GetExtension(field.Options, csipb.E_CsiSecret)
+		if err == nil {
+			_, ok := opt.(*bool)
+			if ok {
+				sanitizeFields = append(sanitizeFields, *field)
+				break
+			}
+		}
+	}
+
+	msg, ok := pb.(proto.Message)
+	if !ok {
+		return ""
+	}
+	var sMsg, sElem string
+	var sanitized bool
+
+	msgValue := reflect.ValueOf(msg)
+	msgElem := msgValue.Elem()
+	msgType := msgElem.Type()
+	for n := 0; n < msgElem.NumField(); n++ {
+		f := msgElem.Field(n)
+		for _, s := range sanitizeFields {
+			fieldName := s.GetName()
+			fieldName = strings.ToUpper(fieldName[:1]) + fieldName[1:]
+			if fieldName == msgType.Field(n).Name {
+				// Need to sanitize
+				m, ok := reflect.Indirect(msgValue).FieldByName(fieldName).Interface().(map[string]string)
```

Comment:
I think you might be able to sanitize the field by calling:
```
reflect.Zero(msgType)
```

Reply:
That would kill the map completely, I was thinking to preserve `key` for debugging purposes

Reply:
@msau42 I think even easier solution would be just to skip field in proto with `csi_secret`, in this case it will be very generic as regardless of a field type it will not be added to the string.
WDYT?